packages/web-app-files/src/views/spaces/Generic.vue

```diff
@@ -82,79 +86,83 @@ import { mapGetters, mapState, mapActions, mapMutations } from 'vuex'
+import FilesViewWrapper from '../../components/FilesViewWrapper.vue'
+import KeyboardActions from '../../components/FilesList/KeyboardActions.vue'
+import ListInfo from '../../components/FilesList/ListInfo.vue'
+import NotFoundMessage from '../../components/FilesList/NotFoundMessage.vue'
+import Pagination from '../../components/FilesList/Pagination.vue'
+import QuickActions from '../../components/FilesList/QuickActions.vue'
+import ResourceTable from '../../components/FilesList/ResourceTable.vue'
+import SideBar from '../../components/SideBar/SideBar.vue'
 import AppLoadingSpinner from 'web-pkg/src/components/AppLoadingSpinner.vue'
 import NoContentMessage from 'web-pkg/src/components/NoContentMessage.vue'
-import NotFoundMessage from '../components/FilesList/NotFoundMessage.vue'
-import ListInfo from '../components/FilesList/ListInfo.vue'
-import Pagination from '../components/FilesList/Pagination.vue'
-import ContextActions from '../components/FilesList/ContextActions.vue'
-import { createLocationSpaces } from '../router'
-import { useResourcesViewDefaults } from '../composables'
-import { defineComponent, unref, computed } from '@vue/composition-api'
-import { move } from '../helpers/resource'
+
+import { VisibilityObserver } from 'web-pkg/src/observer'
+import { ImageDimension, ImageType } from '../../constants'
+import { bus } from 'web-pkg/src/instance'
+import { breadcrumbsFromPath, concatBreadcrumbs } from '../../helpers/breadcrumbs'
+import { createLocationSpaces } from '../../router'
+import { useResourcesViewDefaults } from '../../composables'
+import { defineComponent } from '@vue/composition-api'
+import { move } from '../../helpers/resource'
 import { Resource } from 'web-client'
 import { useGraphClient } from 'web-client/src/composables'
-import { useCapabilityShareJailEnabled, useRouteParam } from 'web-pkg/src/composables'
-import KeyboardActions from '../components/FilesList/KeyboardActions.vue'
-import SideBar from '../components/SideBar/SideBar.vue'
-import FilesViewWrapper from '../components/FilesViewWrapper.vue'
+import { useCapabilityShareJailEnabled } from 'web-pkg/src/composables'
+import { Location } from 'vue-router'
 
 const visibilityObserver = new VisibilityObserver()
 
 export default defineComponent({
+  name: 'GenericSpace',
+
   components: {
-    FilesViewWrapper,
     AppBar,
-    CreateAndUpload,
-    ResourceTable,
-    QuickActions,
     AppLoadingSpinner,
+    ContextActions,
+    CreateAndUpload,
+    FilesViewWrapper,
+    KeyboardActions,
+    ListInfo,
     NoContentMessage,
     NotFoundMessage,
-    ListInfo,
     Pagination,
-    ContextActions,
-    KeyboardActions,
+    QuickActions,
+    ResourceTable,
     SideBar
   },
+
   mixins: [
     MixinAccessibleBreadcrumb,
     MixinFileActions,
     MixinFilesListScrolling,
     MixinFilesListFilter
   ],
+
   props: {
     space: {
       type: Object,
-      required: false
+      required: true
     }
```

Comment:
Does this work with oc10?

Reply:
Or is that relevant because of the legacy view?

Reply:
I want to introduce a kind of "virtual space" object for when spaces are not available in the backend. Currently have the impression that it would reduce quite some code complexity. That being said, it only works for ocis at the moment 😅